sql/src/main/java/io/crate/execution/dsl/phases/JoinPhase.java

```diff
@@ -95,6 +98,8 @@ public JoinPhase(ExecutionPhase.Type type,
         this.numLeftOutputs = numLeftOutputs;
         this.numRightOutputs = numRightOutputs;
         this.leftOutputTypes = leftOutputTypes;
+        Preconditions.checkArgument(leftSize <= Integer.MAX_VALUE, "leftSize must be <= max integer value");
```

Comment:
I think we should throw a more concrete exception here with a better message as this will be bubbled to the user. something like "hash join not possible as the table contains too much rows..."